sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -652,6 +676,13 @@ def collect_metrics(self):
             with self.connection.get_managed_cursor() as cursor:
                 cursor.execute("SET NOCOUNT ON")
             try:
+                # Server state queries require VIEW SERVER STATE permissions, which some managed database
+                # versions do not support.
+                if self.static_info_cache.get(STATIC_INFO_ENGINE_EDITION) not in [
```

Comment:
Would these run when only custom queries are enabled? Could we write a test for this?

Reply:
Actually, these have nothing to do with custom queries now! With this PR: https://github.com/DataDog/integrations-core/pull/11869, they are decoupled from custom queries. So `QueryExecutor` is used purely as a mechanism to execute the queries and submit the metrics. I would like to move the database integrations to this model of fetching metrics.

Reply:
We discussed offline. There is a bug in the `only_custom_queries` option that is out of scope for this PR. We will investigate separately.